### PR TITLE
Fixes #25629: Fix changing disc size

### PR DESCRIPTION
**Pull request by leewaa**
Problem here was the type check for `volume_attrs` against `Hash` when its actually a `HashWithIndifferentAccess`. I made this more solid with `vm_volume.size_gb = volume_attrs[:size_gb] if volume_attrs[:size_gb].present?`. After testing this seems to be fine. We need to implement some happy path feature tests then things like this would be caught much earlier.

### CODE REVIEW

File: app/models/compute_resources/foreman/model/vmware.rb
Comment (ekohl) [inline app/models/compute_resources/foreman/model/vmware.rb:482]: Perhaps it should also check if it's not nil? I assume that why it checked for a hash

Comment (leewaa) [reply app/models/compute_resources/foreman/model/vmware.rb:482]: `if volume_attrs[:size_gb].present?` is good enough as it only returns true if  `volume_attrs[:size_gb]` has a value... 

however I just added a guard against volume_attrs being `blank` just incase.

Comment (ekohl) [reply app/models/compute_resources/foreman/model/vmware.rb:482]: I meant that if `volume_attrs` itself is nil it raise `NoMethodError: undefined method `[]' for nil:NilClass`

**Review by timogoebel — DISMISSED** [review]
Makes sense.

**Comment by timogoebel** [comment]
@leewaa: Can you please add a test for this?

**Comment by chris1984** [comment]
@timogoebel do you want me to fire off the regression test suite against this pr for sanity checking?

**Comment by timogoebel** [comment]
@chris1984: That would be awesome.

**Comment by timogoebel** [comment]
@leewaa: Tests are still missing.

**Comment by leewaa** [comment]
Back from my break, on this now

**Comment by chris1984** [comment]
@timogoebel starting regression tests, was away on holiday

**Comment by leewaa** [comment]
@timogoebel: test is in, please observe the comments in test also

**Comment by timogoebel** [comment]
Looks good to me, @tbrisker: Do you mind taking a look as well? Can we merge this as it's (quite a small) bug fix?

**Comment by chris1984** [comment]
Thanks,  all the regression tests pass and normal testing pass:
```
Image:

Normal VM creation with single HDD on Local Datastore - Passed
Normal VM creation with added HDD on Local Datastore - Passed
Normal VM creation with single HDD on Storage Pod - Passed

PXE:

Normal VM creation with single HDD on Local Datastore - Passed

Normal VM creation with added HDD on Local Datastore - Passed

Normal VM creation with single HDD on Storage Pod - Passed

Normal VM creation with added HDD on Storage Pod - Passed

Bootdisk:

Normal VM creation with added HDD on Local Datastore - Passed

Normal VM creation with single HDD on Storage Pod - Passed

Normal VM creation with added HDD on Storage Pod - Passed


Network:

Normal VM creation with NIC in portgroup on Standard Switch - Passed

Normal VM creation with NIC in portgroup on Distributed Switch - Passed

Normal VM creation with NIC in port group on Distributed Switch with VLAN - Passed

Normal VM creation with additional NIC - Passed
```

**Comment by chris1984** [comment]
[test katello]

File: test/integration/host_js_test.rb
Comment (tbrisker) [inline test/integration/host_js_test.rb]: Shouldn't this be `Fog.unmock!`?

**Comment by leewaa** [comment]
Ah 🙈 yes!

File: test/integration/host_js_test.rb
Comment (leewaa) [reply test/integration/host_js_test.rb]: oops 🙈 thanks!

**Comment by tbrisker** [comment]
@timogoebel feel free to merge this pre-branching once @leewaa fixes the comment. Thanks!

**Comment by leewaa** [comment]
@timogoebel Good to merge.

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Tested, works well.
If I decrease the disk size, I get this error message: `InvalidDeviceOperation: Invalid operation for device '0'.`. Can we make this more clear in a follow-up PR?

Waiting for Mr. Jenkins before merging.

**Review by tbrisker — APPROVED** [review]
Foreman test is green, upgrade and katello failures unrelated. Thanks @leewaa and @timogoebel !

**Pull request merged**
